### PR TITLE
First scenario attempt 

### PR DESCRIPTION
**Pull request by tomaas-zeman**
- New community scenario with necessary assets
- Added table from Homm3BG repo
- New section header that supports multiline - do we want to replace the original `\addsection` command with it?
Screenshot 2024-06-22 at 23 45 56
Screenshot 2024-06-22 at 23 46 43

en

### CODE REVIEW

**Comment by tomaas-zeman** [comment]
en-4
en-5
en-6

**Comment by tomaas-zeman** [comment]
The footer graphics seems to be a bit off, the page number is center correctly

**Comment by tomaas-zeman** [comment]
Maybe we should also ask authors of community scenarios for approval to publish it here

File: sections/sentinels.tex [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  \medskip
+  \textbf{Strength of Enemy Armies}\\
+  \bigskip
+
+  \newcommand{\bronze}[0]{\svg[12]{bronze}}
+  \newcommand{\silver}[0]{\svg[12]{silver}}
+  \newcommand{\golden}[0]{\svg[12]{golden}}
+  \newcommand{\azure}[0]{\svg[12]{azure}}
+
+  \phantom{\bronze\silver\golden\azure}
```
Comment (tomaas-zeman) [inline sections/sentinels.tex:153]: Is there a better way to work with svg than this? I need to use them before `tabularx` or it doesn't work

File: metadata.tex [hunk 2]
```diff
@@ -291,4 +386,6 @@
 
 \clearpage
 
+\include{sections/sentinels.tex}
```
Comment (qwrtln) [inline metadata.tex:389]: This is good for now, I'm just thinking out loud. Maybe we should create separate files with appropriate dirs, like `campaigns.tex`, `clash.tex`, `coop.tex`, etc.

Comment (qwrtln) [reply metadata.tex:389]: Scenarios could be listed in those files in a pre-defined order, while individual scenarios could reside in their own TeX files in appropriate dirs.

File: sections/sentinels.tex [hunk 3]
```diff
@@ -0,0 +1,183 @@
+% !TeX spellcheck = en_US
+\addscenariosection{1}{Cooperative scenario}{Sentinels}{\images/sentinel.png}
```
Comment (qwrtln) [inline sections/sentinels.tex:2]: Wow, this header looks fantastic, nice work 👍🏻

Comment (qwrtln) [reply sections/sentinels.tex:2]: BTW, where is the graphic coming from? Is it safe to use?

File: sections/sentinels.tex [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        & \lightcell[1.4]{\silver \golden \golden \golden \golden} 
+        & \lightcell[1.4]{\golden \golden \golden \golden \azure} 
+        & \lightcell[1.4]{\golden \azure \azure \azure \footref{azure} \linebreak 
+            (7\golden, 3\azure)}\\
+  \end{tabularx}
+}
+
+\subsection*{\MakeUppercase{Credits}}
+
+\textbf{Author:} silence70011
```
Comment (qwrtln) [inline sections/sentinels.tex:179]: I think the author should be at the top.

File: templates/campaign.tex [hunk 5]
```diff
@@ -0,0 +1,100 @@
+% !TeX spellcheck = en_US
+\addscenariosection{1}{Campaign name}{1. Scenario name}{\images/frenzy.pn}
```
Comment (qwrtln) [inline templates/campaign.tex:2]:
```suggestion
\addscenariosection{1}{Campaign name}{1. Scenario name}{\images/frenzy.png}
```

Comment (qwrtln) [reply templates/campaign.tex:2]: I think the image is missing?

File: templates/campaign.tex [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+\textbf{Win:} \textit{Describe winning conditions ...}
+
+\textbf{Lose:} \textit{Describe losing conditions ...}
+
+\subsection*{\MakeUppercase{Timed events}}
+
+\textbf{1$^{st}$ Round:}
+\begin{itemize}
+    \item Read the "First" section.
```
Comment (qwrtln) [inline templates/campaign.tex:78]:
```suggestion
    \item Read the ``First'' section.
```
I know it looks weird, but renders correct quotation marks in accordance to English typography.

File: templates/campaign.tex [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+\end{itemize}
+
+\textbf{When you complete the scenario:}
+\begin{itemize}
+    \item Read the "Last" section.
```
Comment (qwrtln) [inline templates/campaign.tex:83]:
```suggestion
    \item Read the ``Last'' section.
```

File: tools/compare_pages.sh [hunk 8]
```diff
@@ -48,7 +48,7 @@ download_base_file() {
   local language="$1"
   local printable="$2"
   local type=$(file_type "$printable")
-  local url="https://raw.githubusercontent.com/qwrtln/Homm3BG-build-artifacts/${language}/${type}_${language}.pdf"
+  local url="https://raw.githubusercontent.com/qwrtln/Homm3BG-mission-book-build-artifacts/${language}/${type}_${language}.pdf"
```
Comment (qwrtln) [inline tools/compare_pages.sh:51]: Vigilant!

File: templates/campaign.tex (hunk 5 above)
Comment (qwrtln) [reply templates/campaign.tex:2]: Oh, I just noticed this part is not rendered.

**Comment by qwrtln** [comment]
> The footer graphics seems to be a bit off, the page number is center correctly

You're right! Sorry about that, let me fix it.

**Comment by qwrtln** [comment]
> Maybe we should also ask authors of community scenarios for approval to publish it here

@Nohud, you mentioned some authors were excited about that idea. Can you refer their scenarios?

File: metadata.tex (hunk 2 above)
Comment (tomaas-zeman) [reply metadata.tex:389]: Yes, I agree. We can adjust the structure as we include more missions

File: templates/campaign.tex (hunk 5 above)
Comment (tomaas-zeman) [reply templates/campaign.tex:2]: Ha, good catch. I only updated the scenario but not the templates :)

**Comment by qwrtln** [comment]
Why is the middle of the scenario description non-2-columnar? Any reason in particular?

File: sections/sentinels.tex (hunk 3 above)
Comment (tomaas-zeman) [reply sections/sentinels.tex:2]: A little bit of dall-e, a little bit of photoshop :) I was not sure whether to put the high-res image or not.

Original (hi-res)             |  Slightly lower quality
:-------------------------:|:-------------------------:
sentinel  |   sentinel

I wanted to have this oil-paint effect or whatever it is to get the Heroes look & feel. Not sure how much successful I was though

File: templates/campaign.tex (hunk 5 above)
Comment (tomaas-zeman) [reply templates/campaign.tex:2]: Maybe I will remove the templates for now and create a separate PR for it

**Comment by tomaas-zeman** [comment]
> Why is the middle of the scenario description non-2-columnar? Any reason in particular?

It felt slightly better (more readable) since it's a quite large amount of text

en-5

File: sections/sentinels.tex (hunk 3 above)
Comment (qwrtln) [reply sections/sentinels.tex:2]: Looks very good. I don't mind hi-res. Ghostscript does a very good job with overall PDF size compression (it shrunk the rule book from 77 MB to 22 without any loss of quality). As long as we're not approaching GitHub's 100 MB file size limit, I say go for the hi-res.

**Comment by qwrtln** [comment]
> It felt slightly better (more readable) since it's a quite large amount of text

Agreed, the sparse version looks better than condensed 2 columns 👍🏻

File: sections/sentinels.tex (hunk 4 above)
Comment (tomaas-zeman) [reply sections/sentinels.tex]: done

File: sections/sentinels.tex (hunk 3 above)
Comment (tomaas-zeman) [reply sections/sentinels.tex:2]: This one has 1.5MB but it was more about getting the HOMM feel. I put the full quality back until I find a way to create similar graphics

File: sections/sentinels.tex [hunk 9]
```diff
@@ -3,6 +3,10 @@
 
 \begin{multicols*}{2}
 
+\textbf{Author:} silence70011
+
+\textbf{Source:} \href{https://discord.com/channels/740870068178649108/1233112440322002964/1233112440322002964}{Archon Studio Discord}
```
Comment (qwrtln) [inline sections/sentinels.tex:8]: I don't think source is necessary, but in any case it belongs at the bottom.

Comment (tomaas-zeman) [reply sections/sentinels.tex:8]: I wanted to fully credit the guy. But we'll see, I contacted him to get his approval and thoughts on how he wants to received credit

**Comment by qwrtln** [comment]
`sentinel.png` file is quite large, but in reality you're using only a fraction of its size (just in the header). Would you consider shrinking it?

File: sections/sentinels.tex [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    \item On Round 10 a Boss Army spawns and is different from previous armies, because it gets reinforced 
+    from a fixed pool of neutral units (reinforcement pool = number in brackets in the table below).
+    \item At the beginning of the 2$^{nd}$ combat round and every following round 2 units reinforce the
+    remaining army from the shuffled reinforcement pool (up to the maximum of 5 units) to
+    continue the fight until either the player retreats or every neutral unit from the reinforcement
+    pool is killed. \textit{If the number of enemy units at the beginning of a round is 1 or lower, draw up to 
+    a total of 4 units (e.i. there are always 4 or 5 enemy units on the board after reinforcing, if available).}
+    \item Place the reinforcing units on the enemy base line starting from the left with the lowest initiative
+    (\svg{unit_ranged} units first). If necessary continue on the next line (in the same order).
+    \item When a player retreats the enemy army draws up to the "minimum" of 4 units and takes these
```
Comment (qwrtln) [inline sections/sentinels.tex:140]: Quotes

File: sections/sentinels.tex (hunk 9 above)
Comment (qwrtln) [reply sections/sentinels.tex:8]: I wouldn't wait with merging. If the guy decides he doesn't want to be featured here, we'll delete it.

**Review by qwrtln — APPROVED** [review]

**Pull request merged**